File: Easy/344. Reverse String/main.cpp
#include <iostream>
#include <string>
#include <vector>
#include <unordered_map>


using namespace std;

class Solution {
public:
    void reverseString(vector<char>& s) {
        int start_idx = 0;
        int end_idx = s.size()-1;
        char temp = ' ';

        while (start_idx < end_idx)
        {
            temp = s[start_idx];
            s[start_idx] = s[end_idx];
            s[end_idx] = temp;
            start_idx++;
            end_idx--;
        }

    }
};



int main() {


    Solution s = Solution();
    vector<char> t1 = { 'o', 'l', 'l', 'e', 'h' };

    s.reverseString(t1);

    for (auto&& i : t1)
    {
        cout << i << endl;
    }



    return 0;
}

File: Easy/344. Reverse String/main.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: List[str]
        :rtype: None Do not return anything, modify s in-place instead.
        """
        start_idx=0
        end_idx=len(s)-1
        temp=""
        while start_idx<end_idx:
            temp=s[start_idx]
            s[start_idx]=s[end_idx]
            s[end_idx]=temp
            start_idx+=1
            end_idx-=1
    
        
                    
s=Solution()

test1 = ["h","e","l","l","o"]

s.reverseString(test1)
print(test1)
